cmd/algorand-indexer/daemon.go

```diff
@@ -60,265 +64,336 @@ var (
+
+	cfg.flags.StringVarP(&cfg.indexerDataDir, "data-dir", "i", "", "path to indexer data dir, or $INDEXER_DATA")
+	cfg.flags.BoolVar(&cfg.initLedger, "init-ledger", true, "initialize local ledger using sequential mode")
+	cfg.flags.StringVarP(&cfg.catchpoint, "catchpoint", "", "", "initialize local ledger using fast catchup")
+
+	cfg.flags.StringVarP(&cfg.cpuProfile, "cpuprofile", "", "", "file to record cpu profile to")
+	cfg.flags.StringVarP(&cfg.pidFilePath, "pidfile", "", "", "file to write daemon's process id to")
+	cfg.flags.StringVarP(&cfg.configFile, "configfile", "c", "", "file path to configuration file (indexer.yml)")
+	viper.RegisterAlias("algod", "algod-data-dir")
+	viper.RegisterAlias("algod-net", "algod-address")
+	viper.RegisterAlias("server", "server-address")
+	viper.RegisterAlias("token", "api-token")
+	viper.RegisterAlias("data-dir", "data")
+	return daemonCmd
+}
+
+func configureIndexerDataDir(indexerDataDir string) error {
+	var err error
+	if indexerDataDir == "" {
+		err = fmt.Errorf("indexer data directory was not provided")
+		logger.WithError(err).Errorf("indexer data directory error, %v", err)
+		return err
+	}
+	if _, err = os.Stat(indexerDataDir); os.IsNotExist(err) {
+		err = os.Mkdir(indexerDataDir, 0755)
 		if err != nil {
-			fmt.Fprintf(os.Stderr, "failed to configure logger: %v", err)
-			panic(exit{1})
+			logger.WithError(err).Errorf("indexer data directory error, %v", err)
+			return err
 		}
+	}
+	return err
+}
 
-		if indexerDataDir == "" {
-			fmt.Fprint(os.Stderr, "indexer data directory was not provided")
-			panic(exit{1})
+func loadIndexerConfig(indexerDataDir string, configFile string) error {
+	var err error
+	var resolvedConfigPath string
+	indexerConfigAutoLoadPath := filepath.Join(indexerDataDir, autoLoadIndexerConfigName)
+	indexerConfigFound := util.FileExists(indexerConfigAutoLoadPath)
+	if indexerConfigFound {
+		//autoload
+		if configFile != "" {
+			err = fmt.Errorf("indexer configuration was found in data directory (%s) as well as supplied via command line.  Only provide one",
+				indexerConfigAutoLoadPath)
+			logger.Error(err)
+			return err
 		}
-		// Create the directory if it doesn't exist
-		if _, err := os.Stat(indexerDataDir); os.IsNotExist(err) {
-			err := os.Mkdir(indexerDataDir, 0755)
-			maybeFail(err, "failure while creating data directory: %v", err)
+		resolvedConfigPath = indexerConfigAutoLoadPath
+	} else if configFile != "" {
+		// user specified
+		resolvedConfigPath = configFile
+	} else {
+		// neither autoload nor user specified
+		return err
+	}
+	configs, err := os.Open(resolvedConfigPath)
+	if err != nil {
+		logger.WithError(err).Errorf("File Does Not Exist Error: %v", err)
+		return err
+	}
+	defer configs.Close()
+	err = viper.ReadConfig(configs)
+	if err != nil {
+		logger.WithError(err).Errorf("invalid config file (%s): %v", viper.ConfigFileUsed(), err)
+		return err
+	}
+	logger.Infof("Using configuration file: %s\n", resolvedConfigPath)
```

Comment:
After this log statement we should call BindSetFlags() to make sure we get everything set.